kolibri/core/auth/api.py

```diff
@@ -885,7 +901,16 @@ def create(self, request):
             unauthenticated_user = FacilityUser.objects.get(
                 username__exact=username, facility=facility_id
             )
-        user = authenticate(username=username, password=password, facility=facility_id)
+        user = None
+        if interface.enabled and valid_app_key_on_request(request):
+            # If we are in app context, then try to get the automatically created OS User
+            # if it matches the username, without needing a password.
```

Comment:
I'm not sure how the os_user stuff actually is supposed to be working, but this is just to check that there is an operating system user (presumably the one who is currently logged in/the active os user of the device), with the same name as an existing kolibri user. And if so, just proceed using that, otherwise do normal kolibri authentication?

But, does Kolibri use some other sort of credential (like a password or token) in the os_authentication? I'm asking because I'm wondering if there would be a scenario where there is a os user and a kolibri user with the same username, but some other part of the authentication goes wrong (i.e. OS user resets their password?). It's very hypothetical

Reply:
It's not _super_ hypothetical, as due to syncing, you can have multiple facility users created in a facility with the same username.

The os_user look up, however, is not actually based on the username, but the auth token that is encoded in `APP_AUTH_TOKEN_COOKIE_NAME`, so all we are doing here is attempting to see if there is an os user with an active auth token, and if the username of the FacilityUser that resolves to matches the username sent to the backend in the API request. In this case, and this case only, we side step the usual authentication procedure, because this user is 'pre-authenticated' by virtue of the app context providing the auth token.